kv/kvclient/kvcoord: TestDistSenderReplicaStall failed

kv/kvclient/kvcoord.TestDistSenderReplicaStall [failed](https://tanzanite.cluster.engflow.com/invocations/default/11802f68-0204-4eef-8f3d-9a7598f7085e?testReportRun=3&testReportShard=7&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZjbGllbnQva3Zjb29yZDprdmNvb3JkX3Rlc3Q=) on master @ [b3fec61ca90095c664f6432af864f18e9946f8bb](https://github.com/cockroachdb/cockroach/commits/b3fec61ca90095c664f6432af864f18e9946f8bb):


```
    dist_sender_circuit_breaker_test.go:108: Get returned {/Min <nil>} err=aborted in DistSender: result is ambiguous: context deadline exceeded
    dist_sender_circuit_breaker_test.go:104: sending Get request
    dist_sender_circuit_breaker_test.go:108: Get returned {/Min <nil>} err=aborted in DistSender: result is ambiguous: context deadline exceeded
    dist_sender_circuit_breaker_test.go:104: sending Get request
    dist_sender_circuit_breaker_test.go:108: Get returned {/Min <nil>} err=aborted in DistSender: result is ambiguous: context deadline exceeded
    dist_sender_circuit_breaker_test.go:104: sending Get request
    dist_sender_circuit_breaker_test.go:108: Get returned {/Min <nil>} err=aborted in DistSender: result is ambiguous: context deadline exceeded
    dist_sender_circuit_breaker_test.go:104: sending Get request
    dist_sender_circuit_breaker_test.go:108: Get returned {/Min <nil>} err=aborted in DistSender: result is ambiguous: context deadline exceeded
    dist_sender_circuit_breaker_test.go:104: sending Get request
    dist_sender_circuit_breaker_test.go:108: Get returned {/Min <nil>} err=aborted in DistSender: result is ambiguous: context deadline exceeded
    dist_sender_circuit_breaker_test.go:104: sending Get request
    dist_sender_circuit_breaker_test.go:108: Get returned {/Min <nil>} err=aborted in DistSender: result is ambiguous: context deadline exceeded
    dist_sender_circuit_breaker_test.go:104: sending Get request
    dist_sender_circuit_breaker_test.go:108: Get returned {/Min <nil>} err=aborted in DistSender: result is ambiguous: context deadline exceeded
    dist_sender_circuit_breaker_test.go:104: sending Get request
    dist_sender_circuit_breaker_test.go:108: Get returned {/Min <nil>} err=aborted in DistSender: result is ambiguous: context deadline exceeded
    dist_sender_circuit_breaker_test.go:104: sending Get request
    dist_sender_circuit_breaker_test.go:108: Get returned {/Min <nil>} err=aborted in DistSender: result is ambiguous: context deadline exceeded
    dist_sender_circuit_breaker_test.go:104: sending Get request
    dist_sender_circuit_breaker_test.go:108: Get returned {/Min <nil>} err=aborted in DistSender: result is ambiguous: context deadline exceeded
    dist_sender_circuit_breaker_test.go:104: sending Get request
    dist_sender_circuit_breaker_test.go:108: Get returned {/Min <nil>} err=aborted in DistSender: result is ambiguous: context deadline exceeded
    dist_sender_circuit_breaker_test.go:104: sending Get request
    dist_sender_circuit_breaker_test.go:108: Get returned {/Min <nil>} err=aborted in DistSender: result is ambiguous: context deadline exceeded
    dist_sender_circuit_breaker_test.go:104: sending Get request
    dist_sender_circuit_breaker_test.go:108: Get returned {/Min <nil>} err=aborted in DistSender: result is ambiguous: context deadline exceeded
    dist_sender_circuit_breaker_test.go:104: sending Get request
    dist_sender_circuit_breaker_test.go:108: Get returned {/Min <nil>} err=aborted in DistSender: result is ambiguous: context deadline exceeded
    dist_sender_circuit_breaker_test.go:104: sending Get request
    dist_sender_circuit_breaker_test.go:108: Get returned {/Min <nil>} err=aborted in DistSender: result is ambiguous: context deadline exceeded
    dist_sender_circuit_breaker_test.go:104: sending Get request
    dist_sender_circuit_breaker_test.go:108: Get returned {/Min <nil>} err=aborted in DistSender: result is ambiguous: context deadline exceeded
    dist_sender_circuit_breaker_test.go:104: sending Get request
    dist_sender_circuit_breaker_test.go:108: Get returned {/Min <nil>} err=aborted in DistSender: result is ambiguous: context deadline exceeded
    dist_sender_circuit_breaker_test.go:104: sending Get request
    dist_sender_circuit_breaker_test.go:108: Get returned {/Min <nil>} err=aborted in DistSender: result is ambiguous: context deadline exceeded
    dist_sender_circuit_breaker_test.go:104: sending Get request
    dist_sender_circuit_breaker_test.go:108: Get returned {/Min <nil>} err=aborted in DistSender: result is ambiguous: context deadline exceeded
    dist_sender_circuit_breaker_test.go:104: sending Get request
    dist_sender_circuit_breaker_test.go:108: Get returned {/Min <nil>} err=aborted in DistSender: result is ambiguous: context deadline exceeded
    dist_sender_circuit_breaker_test.go:104: sending Get request
    dist_sender_circuit_breaker_test.go:108: Get returned {/Min <nil>} err=aborted in DistSender: result is ambiguous: context deadline exceeded
    dist_sender_circuit_breaker_test.go:102: 
        	Error Trace:	pkg/kv/kvclient/kvcoord/dist_sender_circuit_breaker_test.go:102
        	            				pkg/testutils/subtest.go:23
        	Error:      	Received unexpected error:
        	            	context deadline exceeded
        	Test:       	TestDistSenderReplicaStall/clientTimeout=true
    --- FAIL: TestDistSenderReplicaStall/clientTimeout=true (30.09s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=3</code>
 - <code>shard=7</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #136360 kv/kvclient/kvcoord: TestDistSenderReplicaStall failed [A-kv-client C-bug C-test-failure O-robot P-3 T-kv branch-release-24.2.6-rc]
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDistSenderReplicaStall.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-45131

--- a/pkg/kv/kvclient/kvcoord/dist_sender_circuit_breaker_test.go
+++ b/pkg/kv/kvclient/kvcoord/dist_sender_circuit_breaker_test.go
@@ -45,7 +45,7 @@ func TestDistSenderReplicaStall(t *testing.T) {
 		// The lease won't move unless we use expiration-based leases. We also
 		// speed up the test by reducing various intervals and timeouts.
 		st := cluster.MakeTestingClusterSettings()
-		kvserver.ExpirationLeasesOnly.Override(ctx, &st.SV, true)
+		kvserver.OverrideDefaultLeaseType(ctx, &st.SV, roachpb.LeaseExpiration)
 		kvcoord.CircuitBreakersMode.Override(
 			ctx, &st.SV, kvcoord.DistSenderCircuitBreakersAllRanges,
 		)

